presto-main/src/main/java/com/facebook/presto/sql/planner/DomainTranslator.java

```diff
@@ -685,10 +685,6 @@ private int compareOriginalValueToCoerced(Type originalValueType, Object origina
         @Override
         protected ExtractionResult visitInPredicate(InPredicate node, Boolean complement)
         {
-            if (!(node.getValue() instanceof SymbolReference) || !(node.getValueList() instanceof InListExpression)) {
```

Comment:
most likely you should keep: `!(node.getValueList() instanceof InListExpression)`